tests/next_tests/integration_tests/feature_tests/ffront_tests/ffront_test_utils.py

```diff
@@ -118,18 +119,38 @@ def debug_itir(tree):
 Cell = gtx.Dimension("Cell")
 EdgeOffset = gtx.FieldOffset("EdgeOffset", source=Edge, target=(Edge,))
 
+V2EDim = gtx.Dimension("V2E", kind=gtx.DimensionKind.LOCAL)
+E2VDim = gtx.Dimension("E2V", kind=gtx.DimensionKind.LOCAL)
+C2EDim = gtx.Dimension("C2E", kind=gtx.DimensionKind.LOCAL)
+C2VDim = gtx.Dimension("C2V", kind=gtx.DimensionKind.LOCAL)
+V2E = gtx.FieldOffset("V2E", source=Edge, target=(Vertex, V2EDim))
+E2V = gtx.FieldOffset("E2V", source=Vertex, target=(Edge, E2VDim))
+C2E = gtx.FieldOffset("C2E", source=Edge, target=(Cell, C2EDim))
+C2V = gtx.FieldOffset("C2V", source=Vertex, target=(Cell, C2VDim))
+
 size = 10
 
 
-@pytest.fixture
-def reduction_setup():
+class MeshDescriptor(Protocol):
+    @property
+    def num_vertices(self) -> int: ...
+
+    @property
+    def num_cells(self) -> int: ...
+
+    @property
+    def num_edges(self) -> int: ...
+
+    @property
+    def num_levels(self) -> int: ...
+
+    @property
+    def offset_provider(self) -> dict[str, common.Connectivity]: ...
+
+
+def simple_mesh() -> MeshDescriptor:
```

Comment:
still not a beauty. happy to improve more if you think would be good.

Reply:
It looks fine to me. I guess this concept is the same as the `BaseGrid` in [icon4py](https://github.com/C2SM/icon4py/blob/4cc9eefed579cf66475c6890ea6919182a8bde51/model/common/src/icon4py/model/common/grid/base.py#L56). Ideally they should be aligned so I guess someone should rename the `offset_provider` properties in icon4py to the name used here because their names seem wrong to me.